cli/azd/cmd/infra_create.go

```diff
@@ -259,7 +259,33 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 		return fmt.Errorf("deployment failed: %w", err)
 	}
 
+	dotnetCli := tools.NewDotNetCli()
+	if err := tools.EnsureInstalled(ctx, dotnetCli); err != nil {
+		return err
+	}
+
+	for _, svc := range proj.Services {
+		if svc.Language == "dotnet" {
+			err := dotnetCli.InitializeSecret(ctx, parametersPath)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
 	template.CanonicalizeDeploymentOutputs(&res.Result.Properties.Outputs)
+
+	for _, svc := range proj.Services {
+		{
```

Comment:
NIT: There's an extra set of indentation here, I think you can remove the open/curly braces on L279/286.